client/src/Pos/Client/Txp/Util.hs

```diff
@@ -183,20 +189,21 @@ type TxCreateMode m
 
 -- | Generic function to create a transaction, given desired inputs,
 -- outputs and a way to construct witness from signature data
-makeAbstractTx :: (owner -> TxSigData -> TxInWitness)
+makeAbstractTx :: (owner -> TxSigData -> Either e TxInWitness)
                -> TxOwnedInputs owner
                -> TxOutputs
-               -> TxAux
-makeAbstractTx mkWit txInputs outputs = TxAux tx txWitness
-  where
+               -> Either e TxAux
+makeAbstractTx mkWit txInputs outputs = do
+  let
     tx = UnsafeTx (map snd txInputs) txOutputs txAttributes
     txOutputs = map toaOut outputs
     txAttributes = mkAttributes ()
-    txWitness = V.fromList $ toList $ txInputs <&>
-        \(addr, _) -> mkWit addr txSigData
     txSigData = TxSigData
         { txSigTxHash = hash tx
         }
+  txWitness <- V.fromList . toList <$>
+      for txInputs (\(addr, _) -> mkWit addr txSigData)
```

Comment:
Out loud thought: If I'm not mistaken (I might just well be 😁 ) this will return the very last failure, but I can imagine scenarios where we might want to stop and report the first failure. Nothing to stop this PR from being merged though. I'm I'm being mental, simply ignore 😀

Reply:
The `Either` monad stops on the first failure, check in GHCi:

```
ghci> for [1,2,3] Left
Left 1
```